nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -10,7 +10,8 @@ let
 
   interfaces = attrValues config.networking.interfaces;
 
-  enableDHCP = config.networking.useDHCP || any (i: i.useDHCP == true) interfaces;
+  enableDHCP = config.networking.dhcpcd.enable &&
+        (config.networking.useDHCP || any (i: i.useDHCP == true) interfaces);
```

Comment:
I thought about that aswell, but one _might_ want to setup networkd with dhcpcd and then it would fail..
Though I haven't tested whether networkd with dhcpcd even works...

So the question is whether we want to support `networkd` with `dhcpcd` or not?